src/main/java/com/bgsoftware/superiorskyblock/service/region/ProtectionHelper.java

```diff
@@ -33,4 +46,26 @@ public static boolean shouldPreventInteraction(InteractionResult interactionResu
         throw new IllegalStateException("No handling for result " + interactionResult);
     }
 
+    public static void sendProtectionMessage(CommandSender sender) {
```

Comment:
In most cases (if not all of them), Island was already looked-up by the caller.
Add a nullable island parameter to the method. Only if it's null, look for the island again.

Reply:
> But the shouldPreventInteraction() method is mainly used in ProtectionListener, where the island is not known yet, only in RegionManagerServiceImpl there are islands, so how to connect it? Should I "assign" the island to InteractionResult or how? 😕

In this case, pass null